packages/dom/src/test/dom.js

```diff
@@ -153,4 +159,61 @@ describe( 'DOM', () => {
 			expect( isTextField( document.createElement( 'div' ) ) ).toBe( false );
 		} );
 	} );
+
+	describe( 'nodeContains', createNodeContainsTests( nodeContains ) );
+	describe( 'nodeContainsImplementation', createNodeContainsTests( nodeContainsImplementation ) );
 } );
+
+/**
+ * Creates tests for a nodeContains function matching the signature of our `nodeContains` export.
+ *
+ * @param {Function} nodeContainsFunction The nodeContains implementation to test.
+ *
+ * @return {Function} A function that declares the tests.
+ */
+function createNodeContainsTests( nodeContainsFunction ) {
+	return () => {
+		let parentNode;
+		beforeEach( () => {
+			parentNode = document.createElement( 'div' );
+		} );
+
+		it( 'it returns false when the parent doesn\'t contain the candidate element', () => {
+			const orphanElement = document.createElement( 'p' );
+			expect( nodeContainsFunction( parentNode, orphanElement ) ).toBeFalsy();
+		} );
+
+		it( 'it returns false when the parent doesn\'t contain the candidate text node', () => {
+			const orphanTextNode = document.createTextNode( 'test' );
+			expect( nodeContainsFunction( parentNode, orphanTextNode ) ).toBeFalsy();
+		} );
+
+		it( 'it returns true when the parent contains a candidate element as a child', () => {
```

Comment:
We should have a test:

```js
it( 'returns true when passed node itself', () => {
```

...which I have a feeling will fail as current implemented. See also my concluding notes about us wanting to maintain this.

From specification:

>The contains(other) method, when invoked, must return true if other is an inclusive descendant _[...]_ An inclusive descendant is an object or one of its descendants.

https://dom.spec.whatwg.org/#dom-node-contains
https://dom.spec.whatwg.org/#concept-tree-inclusive-descendant

Reply:
Thanks for catching that! I added a test. Surprisingly, it does work due to using `Node#contains` if the node is an element. I don't love that this is IE-specific fix, but I was keeping my understanding of our browser support in mind and only know about IE's broken implementation.

> See also my concluding notes about us wanting to maintain this.

It's possible I'm not reading well because it's later here, but I don't see direct commentary on wanting to maintain this.

I'd rather add an existing polyfill, but when I searched the NPM registry, all I found were implementations like `nodeContains` and nothing to patch in a working `Node#contains`. It wasn't an exhaustive search, and I'll look again in the morning.

Reply:
> I added a test case for the latter, which fails at the moment.

Thanks!
🤦‍♂️ The ironic thing is that this is related to the bug I wanted to solve in the IE implementation: Not working for non-element nodes.

Thanks for finding a good external polyfill for this. It's good to now know about that library.

I updated this PR to add the polyfill before the `'wp-dom'` script since appears to be a foundational script for the packages (even for `wp-element` through its `wp-utils` dependencies). I believe this may fall short of how widely you wanted to load this, but what do you think?